### ISSUE TITLE
[0.2.x-preview] no exception is raised when parsing a set with duplicated values

### ISSUE THREAD

**Issue by gontard** [scalapb/scalapb-validate#58]
When `unique_to_set` is true and a repeated field has the constraint `(validate.rules).repeated.unique = true,` no exception is raised when parsing duplicated values.

Demo in 
https://github.com/gontard/scalapb-validate-demo/commit/dab86bdec8e056517b612995809f32abe5ee19db

**Comment by thesamet** [comment]
Yes, this has also been noted in [https://github.com/scalapb/ScalaPB/commit/4e1fa59970c8c7df707f0accab577a6d8f0160b5#diff-6149aa8f05e4ed90504f7f76bd58033f3c045f5dc21f53dda4dbc2a63e1a15b0R227-R233 should certainly be fixed. This was noted in [https://scalapb.github.io/docs/validation](the documentation) and should certainly be fixed.
> Validation is performed only after a message is instantiated. Therefore, when you use a Set or a NonEmptySet instead of list, duplicate elements are deduped during parsing time, when ScalaPB adds them to the collections' builder. As a result, uniqueness validation will always succeed, even when the original message had duplicate elements.

**Comment by thesamet** [comment]
This should be addressed in 0.2.0-preview4. The change also requires updating ScalaPB's compilerplugin to 0.2.0-preview4 and regenerating all code.

**Comment by thesamet** [comment]
Closing as this is implemented in the preview. Feel free to open another issue if there are any gaps found.
